### PR TITLE
Reworded `build` help and added debug prints

### PR DESCRIPTION
**Pull request by abitrolly**
Debug prints are needed for troubleshooting ARCH selection problem in
https://github.com/openSUSE/osc/pull/241#issuecomment-1203463612

### CODE REVIEW

File: osc/commandline.py [hunk 1]
```diff
@@ -6621,12 +6624,11 @@ def parse_repoarchdescr(self, args, noinit = False, alternative_project = None,
...
 
-        You need to call the command inside a package directory, which should be a
-        buildsystem checkout. (Local modifications are fine.)
+        The command works from a package checkout (local changes are fine).
 
-        The arguments REPOSITORY and ARCH can be taken from the first two columns
-        of the 'osc repos' output. BUILD_DESCR is either a RPM spec file, or a
-        Debian dsc file.
+        You can use `osc repos` to look up REPOSITORY and ARCH arguments. If
+        they are not set, `osc` choses suitable at random. BUILD_DESCR is
```
Comment (hramrach) [inline osc/commandline.py:6630]: I don't think it's good to say the repository is picked at random.

The order of repositories is deterministic, it's just not documented how the repositories are ordered, specifically.

Comment (abitrolly) [reply osc/commandline.py]: Fair enough. Documented the exact order.

File: osc/commandline.py [hunk 2]
```diff
@@ -6380,6 +6380,11 @@ def parse_repoarchdescr(self, args, noinit = False, alternative_project = None,
             arg_repository, arg_arch, arg_descr = args
 
         arg_arch = arg_arch or osc.build.hostarch
+        if self.options.debug:
+            print("hostarch: ", osc.build.hostarch)
+            print("arg_arch: ", arg_arch)
+            print("arg_repository: ", arg_repository)
+            print("arg_descr: ", arg_descr)
```
Comment (dmach) [inline osc/commandline.py:6387]: How about to print debug messages to stderr?
In my opinion: debug belongs to stderr, verbose increases regular output verbosity and belongs to stdout.

Comment (abitrolly) [reply osc/commandline.py:6387]: I stole that part from here.

https://github.com/openSUSE/osc/blob/43337afdf5aa258267d407fdaf59feb3d0b15a78/osc/commandline.py#L2461-L2464

I'll make a helper to make it easier.

Comment (abitrolly) [reply osc/commandline.py]: Added `self_debug` helper, which prints to stderr.

**Comment by dmach** [comment]
Looks good to me, merging.
Thanks!

**Pull request merged**